bindings/apple/build_crypto_xcframework.sh

```diff
@@ -22,52 +38,68 @@ TARGET_CRATE=matrix-sdk-crypto-ffi
+if ${only_ios}; then
+  # iOS
+  echo -e "Building only for iOS"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "aarch64-apple-ios"
+else
+  # iOS
+  echo -e "Building for iOS [1/5]"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "aarch64-apple-ios"
+
+  # MacOS
+  echo -e "\nBuilding for macOS (Apple Silicon) [2/5]"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "aarch64-apple-darwin"
+  echo -e "\nBuilding for macOS (Intel) [3/5]"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "x86_64-apple-darwin"
+
+  # iOS Simulator
+  echo -e "\nBuilding for iOS Simulator (Apple Silicon) [4/5]"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "aarch64-apple-ios-sim"
+  echo -e "\nBuilding for iOS Simulator (Intel) [5/5]"
+  cargo build -p ${TARGET_CRATE} ${REL_FLAG} --target "x86_64-apple-ios"
+fi
 
 echo -e "\nCreating XCFramework"
 # Lipo together the libraries for the same platform
 
-# MacOS
-lipo -create \
-  "${TARGET_DIR}/x86_64-apple-darwin/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
-  "${TARGET_DIR}/aarch64-apple-darwin/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
-  -output "${GENERATED_DIR}/macos/libmatrix_sdk_crypto_ffi.a"
-
-# iOS Simulator
-lipo -create \
-  "${TARGET_DIR}/x86_64-apple-ios/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
-  "${TARGET_DIR}/aarch64-apple-ios-sim/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
-  -output "${GENERATED_DIR}/simulator/libmatrix_sdk_crypto_ffi.a"
-
-# Generate uniffi files
-cargo uniffi-bindgen generate \
-  --language swift \
-  --lib-file "${TARGET_DIR}/aarch64-apple-ios-sim/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
-  --config "${SRC_ROOT}/bindings/${TARGET_CRATE}/uniffi.toml" \
-  --out-dir ${GENERATED_DIR} \
-  "${SRC_ROOT}/bindings/${TARGET_CRATE}/src/olm.udl"
+if ! ${only_ios}; then
+  echo "Lipo together the libraries for the same platform"
+  # MacOS
+  lipo -create \
+    "${TARGET_DIR}/x86_64-apple-darwin/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    "${TARGET_DIR}/aarch64-apple-darwin/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    -output "${GENERATED_DIR}/macos/libmatrix_sdk_crypto_ffi.a"
+
+  # iOS Simulator
+  lipo -create \
+    "${TARGET_DIR}/x86_64-apple-ios/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    "${TARGET_DIR}/aarch64-apple-ios-sim/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    -output "${GENERATED_DIR}/simulator/libmatrix_sdk_crypto_ffi.a"
+fi
+
+if ${only_ios}; then
+  # Generate uniffi files
+  cd ../matrix-sdk-crypto-ffi && cargo run --bin matrix_sdk_crypto_ffi generate \
+    --language swift \
+    --library "${TARGET_DIR}/aarch64-apple-ios/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    --out-dir ${GENERATED_DIR}
+else
+  # Generate uniffi files
+  cd ../matrix-sdk-crypto-ffi && cargo run --bin matrix_sdk_crypto_ffi generate \
+    --language swift \
+    --library "${TARGET_DIR}/aarch64-apple-ios-sim/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
+    --out-dir ${GENERATED_DIR}
+fi
```

Comment:
just not super sure about this part, we are not creating ffi files for Mac in te non iOS case?

Reply:
The previous script was only doing that by default
```
  # Generate uniffi files
  cd ../matrix-sdk-crypto-ffi && cargo run --bin matrix_sdk_crypto_ffi generate \
    --language swift \
    --library "${TARGET_DIR}/aarch64-apple-ios-sim/${REL_TYPE_DIR}/libmatrix_sdk_crypto_ffi.a" \
    --out-dir ${GENERATED_DIR}
```

Do you know what it should be doing?

Reply:
As discussed with Doug, the `if` is not needed. We can just use aarch64-apple-ios for uniffi (built in the 2 options), it will be the same output for all targets.